src/TagsInput/TagsInput.js

```diff
@@ -11,156 +16,163 @@ import InputField from './components/InputField';
+  render() {
+    const {
+      disabled,
+      fontSize,
+      label,
+      onKeyPress,
+      tags,
+      autofocus,
+    } = this.props;
+    const { isFocus, value } = this.state;
+
+    const isButtonDisabled = (!value && tags.length === 0) ?? disabled;
+
+    const dynamicStyle = {
       deleteButton: {
-        opacity: value.length > 0 ? 1 : 0,
+        opacity: value || tags.length > 0 ? 1 : 0,
       },
       label: {
         fontSize,
       },
       border: {
-        web: {
-          boxShadow: isFocus
-            ? `${defaultTokens.borderColorInputFocus} 0 0 0 2px inset`
-            : `${defaultTokens.borderColorInput} 0 0 0 1px inset`,
-        },
+        ...boxShadow(isFocus),
       },
-    });
+    };
 
     return (
-      <View style={[styles.container, dynamicStyle.border, dynamicStyle.label]}>
-        <Text weight="bold" style={styles.label}>
-          {label}
-        </Text>
-        <View style={styles.fieldContainer} onLayout={this.setContainerWidth}>
-          <TagsContainer
-            minWidth={TAGS_MIN_WIDTH}
-            maxWidth={this.subtractFromContainerWidth(inputWidth)}
-            tags={selected}
-            fontSize={fontSize}
-          />
-          <InputField
-            onFocus={this.handleOnFocus}
-            onBlur={this.handleOnBlur}
-            ref={this.inputRef}
-            fontSize={fontSize}
-            disabled={disabled}
-            minWidth={INPUT_MIN_WIDTH}
-            maxWidth={this.subtractFromContainerWidth(TAGS_MIN_WIDTH)}
-            onLayout={this.setInputWidth}
-            value={value}
-            placeholder={this.getPlaceholder()}
-            onChangeText={this.handleChange}
-            autoFocus={isFocus}
+      <TouchableWithoutFeedback onPress={this.handleOnFocus}>
+        <View style={[styles.container, dynamicStyle.border]}>
+          {Boolean(label) && (
+            <Text weight="bold" style={[styles.label, dynamicStyle.label]}>
+              {label}
+            </Text>
+          )}
+          <ScrollView
+            contentContainerStyle={{ flexGrow: 1 }}
+            onContentSizeChange={this.scrollRef?.current?.scrollToEnd}
+            scrollEventThrottle={3}
+            showsHorizontalScrollIndicator={false}
+            horizontal
+            style={styles.fieldContainer}
+            ref={this.scrollRef}
+          >
+            <TagsContainer tags={tags} fontSize={fontSize} />
+            <InputField
+              autofocus={autofocus}
+              onKeyPress={onKeyPress}
+              onFocus={this.handleOnFocus}
+              onBlur={this.handleOnBlur}
+              ref={this.inputRef}
+              fontSize={fontSize}
+              disabled={disabled}
+              value={value}
+              placeholder={this.getPlaceholder()}
+              onChangeText={this.handleChange}
+            />
+          </ScrollView>
+          <DeleteButton
+            opacity={dynamicStyle.deleteButton.opacity}
```

Comment:
This looks rather complex. I believe `style` would have been easier, because you could write
`style={dynamicStyle.deleteButton}`.
Besides, your styles don't need to be fully dynamic:
`style={[style.opacityZero, tags.length > 0 && style.opacityOne, value && dynamicStyle.deleteButton]}`

Reply:
> `style={dynamicStyle.deleteButton}`.
Besides, your styles don't need to be fully dynamic:
`style={[style.opacityZero, tags.length > 0 && style.opacityOne, value && dynamicStyle.deleteButton]}`

💯